packages/backend-core/src/middleware/tests/contentSecurityPolicy.spec.ts

```diff
@@ -132,4 +132,49 @@ describe("contentSecurityPolicy middleware", () => {
     expect(app.getAppMetadata).toHaveBeenCalledWith(appId)
     expect(next).toHaveBeenCalled()
   })
+
+  it("should filter out invalid domains", async () => {
```

Comment:
Should we instead have some integration tests? Make an actual request and verify the header?

Reply:
Do you not think this is good enough? It's hitting the exact same logic and verifying the header set in the koa context. Either way though, I'd like to get this important fix out in the meantime and I can investigate integration tests separately if you think they're needed as well.

Reply:
It'll do for the time being but generally we've been moving away from tests like this on Firestorm. Mocking almost everything is very fragile and often ends up masking real problems. I don't know to what extent we'd have caught this CSP problem with integration tests, probably not at all, but generally I like to hit as much real code in tests as possible.